crates/ln-dlc-node/examples/fund.rs

```diff
@@ -0,0 +1,223 @@
+
+    open_channel(
+        &node,
+        Amount::ONE_BTC
+            .checked_div(10)
+            .expect("small integers to divide"),
+    )
+    .await?;
+
+    let lnd_channels = get_text(&format!("{FAUCET}/lnd/v1/channels")).await?;
+    tracing::info!("open LND channels: {}", lnd_channels);
+    Ok(())
+}
+
+#[derive(Deserialize)]
+struct LndAddr {
+    address: String,
+}
+
+// Includes some bespoke text processing that ensures we can deserialise the response properly
+async fn get_coordinator_address() -> Result<String> {
+    Ok(get_text(&format!("{COORDINATOR}/api/newaddress"))
+        .await?
+        .strip_prefix('"')
+        .to_owned()
+        .unwrap()
+        .strip_suffix('"')
+        .unwrap()
+        .to_owned())
+}
+
+async fn get_text(url: &str) -> Result<String> {
+    Ok(reqwest::get(url).await?.text().await?)
+}
+
+#[derive(Deserialize, Debug)]
+struct BitcoindResponse {
+    result: String,
+}
+
+async fn fund(address: &str, amount: Amount) -> Result<Response> {
+    post_query(
+        "bitcoin",
+        format!(
+            r#"{{"jsonrpc": "1.0", "method": "sendtoaddress", "params": ["{}", "{}"]}}"#,
+            address,
+            amount.to_btc()
+        ),
+    )
+    .await
+}
+
+/// Instructs `bitcoind` to generate to address.
+async fn mine(n: u16) -> Result<()> {
+    let response = post_query(
+        "bitcoin",
+        r#"{"jsonrpc": "1.0", "method": "getnewaddress", "params": []}"#.to_string(),
+    )
+    .await?;
+    let response: BitcoindResponse = response.json().await?;
+
+    post_query(
+        "bitcoin",
+        format!(
+            r#"{{"jsonrpc": "1.0", "method": "generatetoaddress", "params": [{}, "{}"]}}"#,
+            n, response.result
+        ),
+    )
+    .await?;
+    // For the mined blocks to be picked up by the subsequent wallet syncs
+    tokio::time::sleep(Duration::from_secs(5)).await;
+
+    Ok(())
+}
+
+async fn post_query(path: &str, body: String) -> Result<Response> {
+    let client = reqwest::Client::new();
+    let response = client
+        .post(format!("{FAUCET}/{path}"))
+        .body(body)
+        .send()
+        .await?;
+
+    if !response.status().is_success() {
+        bail!(response.text().await?)
+    }
+    Ok(response)
+}
+
+/// Instructs lnd to open a public channel with the target node.
+/// 1. Connect to the target node.
+/// 2. Open channel to the target node.
+async fn open_channel(node_info: &NodeInfo, amount: Amount) -> Result<()> {
+    let port = node_info.address.port();
+    let ip_address = local_ip()?;
+    let host = format!("{ip_address}:{port}");
+    tracing::info!("Connecting lnd to {host}");
+    post_query(
+        "lnd/v1/peers",
+        format!(
+            r#"{{"addr": {{ "pubkey": "{}", "host": "{host}" }}, "perm":false }}"]"#,
+            node_info.pubkey
+        ),
+    )
+    .await?;
+
+    tokio::time::sleep(Duration::from_secs(5)).await;
```

Comment:
Better would be to query lnd for the known peers: https://lightning.engineering/api-docs/api/lnd/lightning/list-peers#rest